### DISCUSSION TITLE
Flattened database views of document data

### DISCUSSION THREAD

**Discussion by mdissel** [JasperFx/marten#2114] [Ideas]
I like the idea of storing json documents and simplify the development. But what about the reporting part of an application? In our sitation the database is also used for read-only direct query access for reporting purposes, mostly done by people with a relational background.

I'm thinking of flattening the json documents to a relational view using sql views. Anyone tested the performance overhead of flattening json documents into a sql view (dynamically created based on the model)?
example:
```
company: [ { 
 id: 1,
 name: 'Company 1',
 addresses: [ { address: 'Street1', city: 'City', ..},{..}]
}, {...}
]

```
would result in two views (i left out the json syntax)
```
select id, name from company 
select address, city, companyId from company.addresses
```

**Comment by jamesfarrer** [comment]
How 'live' does the data need to be for reporting purposes? Perhaps using a materialized view and refreshed every few minutes would be better?

You can then also add indexes on your materialized view and have everything performant
